PolyEngine/Engine/Src/ECS/World.hpp

```diff
@@ -121,16 +121,61 @@ namespace Poly {
 				return **this;
 			}
 
-			ComponentIterator& operator++() { ++primary_iter; return *this; }
-			ComponentIterator operator++(int) { ComponentIterator ret(primary_iter); ++primary_iter; return ret; }
-			ComponentIterator& operator--() { --primary_iter; return *this; }
-			ComponentIterator operator--(int) { ComponentIterator ret(primary_iter); --primary_iter; return ret; }
+			ComponentIterator& operator++() { Increment(); return *this; }
+			ComponentIterator operator++(int) { ComponentIterator ret(primary_iter); Increment(); return ret; }
+			ComponentIterator& operator--() { Decrement(); return *this; }
+			ComponentIterator operator--(int) { ComponentIterator ret(primary_iter); Decrement(); return ret; }
+
+		private:
+			template<int zero = 0>
```

Comment:
Isn't it possible to remove this whole line? It should work.

Reply:
It has to be here, otherwise compiler won't distinguish these two templates.